channeldb/db.go

```diff
@@ -1337,9 +1381,11 @@ func fetchHistoricalChanBucket(tx kvdb.RTx,
 
 // FetchHistoricalChannel fetches open channel data from the historical channel
 // bucket.
-func (d *DB) FetchHistoricalChannel(outPoint *wire.OutPoint) (*OpenChannel, error) {
+func (c *ChannelStateDB) FetchHistoricalChannel(outPoint *wire.OutPoint) (
```

Comment:
Ok I see what type of refactors you were referring to now.

Reply:
I think we could really do more work re: separating things in channeldb. The idea would be to hide things behind intefaces in the end and never allow end-user code to access these buckets at all. This would also allow us to gradually refactor things to work better with other backends.